### PR TITLE
fix(minimax-h3): align Ref2AV output 

### PR DESCRIPTION
**Pull request by Watebear**

### CODE REVIEW

File: lightx2v/common/ops/attn/dynamic_sparse_attn.py
Comment (Watebear) [inline lightx2v/common/ops/attn/dynamic_sparse_attn.py:1]: 这里应该是sparse和packed之后的计算不兼容，所以临时用for循环来替代了。效率肯定低。spare先这样么？

File: lightx2v/common/ops/attn/sol_attn.py
Comment (Watebear) [inline lightx2v/common/ops/attn/sol_attn.py:1]: 这里也是for循环做的兼容

File: lightx2v/common/ops/attn/torch_sdpa.py
Comment (Watebear) [inline lightx2v/common/ops/attn/torch_sdpa.py:1]: for 循环兼容

File: lightx2v/common/ops/mm/mm_weight.py
Comment (Watebear) [inline lightx2v/common/ops/mm/mm_weight.py:1]: tp通信的fp32，不能用 sensetive layer

File: lightx2v/models/audio_encoders/hf/minimax_h3/audio_vae.py
Comment (Watebear) [inline lightx2v/models/audio_encoders/hf/minimax_h3/audio_vae.py:1]: 计算模式对齐

File: lightx2v/models/audio_encoders/hf/minimax_h3/audio_vae.py [hunk 1]
```diff
@@ -62,6 +88,14 @@ def _wn_conv1d(*args, **kwargs) -> nn.Module:
     return weight_norm(nn.Conv1d(*args, **kwargs))
 
 
+@torch.jit.script
+def _snakebeta(hidden_states: torch.Tensor, alpha: torch.Tensor, beta: torch.Tensor) -> torch.Tensor:
```
Comment (Watebear) [inline lightx2v/models/audio_encoders/hf/minimax_h3/audio_vae.py:92]: 这是个优化点，对一个多次调用的激活，做了compile的优化

File: lightx2v/models/input_encoders/hf/minimax_h3/qwen3vl_vision.py [hunk 2]
```diff
@@ -80,14 +114,40 @@ def forward(self, pixels):
...
+        if total_num_heads % tp_size:
+            raise ValueError(f"Qwen3-VL vision heads ({total_num_heads}) must be divisible by TP size ({tp_size})")
+        self.num_heads = total_num_heads // tp_size
+        self.head_dim = config["hidden_size"] // total_num_heads
         self.scaling = self.head_dim**-0.5
         self.qkv = nn.Linear(config["hidden_size"], config["hidden_size"] * 3, bias=True)
         self.proj = nn.Linear(config["hidden_size"], config["hidden_size"], bias=True)
 
+    def shard_for_tensor_parallel(self):
+        if self.tp_size == 1:
```
Comment (Watebear) [inline lightx2v/models/input_encoders/hf/minimax_h3/qwen3vl_vision.py:133]: 配合fp32 tp通信

File: lightx2v/common/ops/rope/h3_sgl_rope.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+            ROPE_DIM=rotary_dim,
+            BLOCK_HALF=triton.next_power_of_2(rotary_dim // 2),
+            IS_BF16=q.dtype is torch.bfloat16,
+            num_warps=1,
+        )
+    return q.reshape(q_shape), k.reshape(k_shape)
+
+
+@ROPE_REGISTER("h3_sgl_rope")
+class MiniMaxH3SGLRope(RopeTemplate):
```
Comment (Watebear) [inline lightx2v/common/ops/rope/h3_sgl_rope.py]: 看能否通用

File: lightx2v/models/schedulers/minimax_h3/scheduler.py [hunk 4]
```diff
@@ -149,18 +183,25 @@ def step_pre(self, step_index):
...
-            # Match MiniMaxH3T2AVDmdTrainer.run_back_simulation exactly.
+        if sampling_profile == "dmd":
             return sample.float() + (sigma - sigma_next) * model_output.float()
+
+        # Rebuild sigma from the timestep to preserve the base model's
+        # reference rounding and operation order.
+        sigma_from_timestep = 1.0 - timestep.to(device=sample.device, dtype=sample.dtype)
         ratio = sigma_next / sigma
-        return ratio * sample.float() + (1.0 - ratio) * denoised.float()
+        state = sample.float()
```
Comment (Watebear) [inline lightx2v/models/schedulers/minimax_h3/scheduler.py:194]: 看输出差异
